src/main/webapp/go.js

```diff
@@ -198,14 +251,15 @@ function updateToCurrentState(index) {
   if (index <= -1) { // The user has not yet pressed the start button.
     toggleHintButton(/* hide = */ true);
   } else { // The user has started the hunt, and needs to solve the riddle.
-    toggleHintButton(/* hide = */ false);
+    delayHintButton();
     hideStartButton();
-    updateMessage(RIDDLE_DISPLAY, 'Riddle: ' + huntArr[index].puzzle);
+    updateMessage(RIDDLE_DISPLAY, 'Riddle: ' + hunt.getDest(index).puzzle);
```

Comment:
Will index be the same as destIndex which is available to the ScavengerHuntManager?  Because then it could just be hunt.getPuzzle() or something